pkg/ottl/functions.go

```diff
@@ -32,65 +55,67 @@ func (p *Parser[K]) newFunctionCall(inv invocation) (Expr[K], error) {
 	if !ok {
 		return Expr[K]{}, fmt.Errorf("undefined function %v", inv.Function)
 	}
-	args, err := p.buildArgs(inv, reflect.TypeOf(f))
+	args := f.CreateEmptyArguments()
+
+	// Pointer values are necessary to fulfill the Go reflection
+	// settability requirements. Non-pointer values are not
+	// modifiable through reflection.
+	if reflect.TypeOf(args).Kind() != reflect.Pointer {
+		return Expr[K]{}, fmt.Errorf("factory for %s must return a pointer to an Arguments value in its CreateEmptyArguments method", inv.Function)
```

Comment:
I like this solution.  Simplification for us, and will be a fast error for function creators if they do it wrong.